File: src/WorkBC.Data/Model/JobBoard/GeocodedLocationCache.cs
using System;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace WorkBC.Data.Model.JobBoard
{
    public class GeocodedLocationCache
    {
        [Key]
        [DatabaseGenerated(DatabaseGeneratedOption.Identity)]
        public int Id { get; set; }

        [StringLength(120)]
        public string Name { get; set; }

        [StringLength(25)]
        public string Latitude { get; set; }

        [StringLength(25)]
        public string Longitude { get; set; }

        [StringLength(80)]
        public string City { get; set; }

        [StringLength(80)]
        public string FrenchCity { get; set; }

        [StringLength(2)]
        public string Province { get; set; }

        public DateTime DateGeocoded { get; set; }

        public bool IsPermanent { get; set; }
    }
}

File: jobboard-laravel/app/Models/GeocodedLocationCache.php
<?php

namespace App\Models;

use Illuminate\Database\Eloquent\Model;

/**
 * Maps the EXISTING `GeocodedLocationCache` table (data-model.md §6; unique
 * `Name`). It is a cache of geocoded locations written by the existing pipeline
 * and the legacy C# GeocodingService.
 *
 * Map, don't create (copilot-instructions): existing PascalCase schema, no
 * timestamps, string PK-less identity `Id`. This app only READS the cache for
 * radius search (Rule B); it never creates/alters the table. The Google Maps
 * write-back path (which WOULD insert rows) is deferred — see {@see
 * \App\Services\Integration\CachedGeocoder}.
 */
final class GeocodedLocationCache extends Model
{
    protected $table = 'GeocodedLocationCache';

    protected $primaryKey = 'Id';

    public $timestamps = false;

    protected $casts = [
        'DateGeocoded' => 'datetime',
        'IsPermanent' => 'bool',
    ];
}
